processor.go

```diff
@@ -170,6 +170,13 @@ func (p *processor) handle(ctx *fh.RequestCtx) {
 		return
 	}
 
+	if p.cfg.Tenant.PrefixPreferSource {
+		sourceTenantPrefix := string(ctx.Request.Header.Peek(p.cfg.Tenant.Header))
+		if sourceTenantPrefix != "" {
+			p.cfg.Tenant.Prefix = sourceTenantPrefix + "-"
```

Comment:
Problem here is that `p.cfg.Tenant.Prefix` is a shared variable. If you'll set it from concurrent connections it'll conflict and probably would be corrupt at some point when two threads would write to it at the same time.

Reply:
OK, I'll see if I can get this scoped properly.